cnf-tests/testsuites/pkg/junit-merger/junit-merger.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"encoding/xml"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/onsi/ginkgo/v2/reporters"
+)
+
+func main() {
+	output := flag.String("output", "-", "The output file name for the merged junit, defaults to stdout (-)")
+
+	// Parse command input
+	flag.Parse()
+
+	junitFiles := flag.Args()
+
+	if len(junitFiles) == 0 {
+		panic("No JUnit files to merge provided.")
+	}
+
+	suites, err := loadJUnitFiles(junitFiles)
+	if err != nil {
+		panic(fmt.Sprintf("Could not load JUnit files. %s", err))
+	}
+
+	result := mergeJUnitFiles(suites)
+
+	writer, err := prepareOutputWriter(*output)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to prepare the output file. %s", err))
+	}
+
+	err = writeJUnitFile(writer, result)
+	if err != nil {
+		panic(fmt.Sprintf("Failed to write the merged junit report. %s", err))
+	}
+}
+
+func loadJUnitFiles(fileGlobs []string) ([]reporters.JUnitTestSuites, error) {
+	result := []reporters.JUnitTestSuites{}
+	for _, fileglob := range fileGlobs {
+		fileglob = filepath.Clean(fileglob)
+		files, err := filepath.Glob(fileglob)
+		if err != nil {
+			return nil, err
+		}
+		for _, file := range files {
+			f, err := os.Open(file)
```

Comment:
Files open seems not to be closed.

Reply:
oops! my bad! thanks!

Reply:
you need to wrap the for body in a func and defer, or close in any return point otherwise you'll leave with open file in case of errors